tests/test_platforms_dummy.py

```diff
@@ -40,15 +42,20 @@ def test_dummy_single_sweep(parameter, average, nshots):
         sweeper = Sweeper(parameter, parameter_range, qubits=[platform.qubits[0]])
     else:
         sweeper = Sweeper(parameter, parameter_range, pulses=[pulse])
-    results = platform.sweep(sequence, sweeper, average=average, nshots=nshots)
+    options = ExecutionParameters(
+        nshots=nshots,
+        averaging_mode=average,
+    )
+    average = not options.averaging_mode is AveragingMode.SINGLESHOT
```

Comment:
Since this is happening many times, you could define a property: 

```py
class ExecutionParameters:
    @property
    def average(self):
        return not self.averaging_mode is AveragingMode.SINGLESHOT
```

Reply:
I will suggest it will be better to recheck this on the dummy platform with @andrea-pasquale tomorrow as average is only needed for the tests in the results part. This is because it uses the old ExecutionResults, we may want to remove when all the platform are updated. With the new ones this should not be needed so we can try to change them.

Reply:
I have the same exact line in the rfsoc driver... mainly because it was easier to convert from the old system to this